File: GXpert/GXpert.Web/Modules/Masters/BloomsTaxanomy/BloomsTaxanomyColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.BloomsTaxanomy")]
[BasedOnRow(typeof(BloomsTaxanomyRow), CheckNames = true)]
public class BloomsTaxanomyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string CoginitiveSkill { get; set; }
    public int SortOrder { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/BloomsTaxanomyColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BloomsTaxanomyRow } from "./BloomsTaxanomyRow";

export interface BloomsTaxanomyColumns {
    Id: Column<BloomsTaxanomyRow>;
    CoginitiveSkill: Column<BloomsTaxanomyRow>;
    SortOrder: Column<BloomsTaxanomyRow>;
}

export class BloomsTaxanomyColumns extends ColumnsBase<BloomsTaxanomyRow> {
    static readonly columnsKey = 'Masters.BloomsTaxanomy';
    static readonly Fields = fieldsProxy<BloomsTaxanomyColumns>();
}